test/common/thread_local/thread_local_impl_test.cc

```diff
@@ -82,6 +82,32 @@ TEST_F(ThreadLocalInstanceImplTest, All) {
   tls_.shutdownThread();
 }
 
+// Validate ThreadLocal::runOnAllThreads behavior with all_thread_complete call back.
+TEST_F(ThreadLocalInstanceImplTest, RunOnAllThreads) {
+
+  SlotPtr tlsptr = tls_.allocateSlot();
+
+  EXPECT_CALL(thread_dispatcher_, post(_));
+  EXPECT_CALL(main_dispatcher_, post(_));
+
+  // Ensure that the thread local call back and all_thread_complete call back are called.
+  std::shared_ptr<std::atomic<bool>> all_threads_complete =
+      std::make_shared<std::atomic<bool>>(false);
+  std::shared_ptr<std::atomic<uint64_t>> thread_local_calls =
```

Comment:
I don't think this needs to be shared; it's owned by the test method.

Reply:
I think earlier we ran in to issue if it is not shared with TSAN build, because multiple thread update it may be. That is why we changed it to shared.

Reply:
I think that's because before you put the condvar in, the test method could exit while threads were active.  The condvar solves that properly, so you shouldn't need to share it anymore.